File: dart/lib/src/repl.dart
import 'dart:async';
import 'dart:convert';
import 'dart:io';

import 'package:dart_deez/dart_deez.dart';

class REPL {
  REPL({
    String? prompt,
  }) : _prompt = prompt ?? '>> ';

  Future<void> run() async {
    stdout
      ..writeln('Welcome to the Monkey Programming Language')
      ..writeln('Feel free to type in commands');

    try {
      // keep reading the stdin until you receive an 'exit' command
      while (true) {
        stdout.write(_prompt);
        final line = stdin.readLineSync(encoding: utf8);
        switch (line?.trim().toLowerCase()) {
          case null:
            break;
          case 'exit':
            stdout.writeln('Bye!');
            exit(0);
          default:
            final lexer = Tokenizer(line!);
            final tokens = <Token>[];
            var token = lexer.nextToken();
            while (token.type != TokenType.eof) {
              tokens.add(token);
              token = lexer.nextToken();
            }
            for (final element in tokens) {
              stdout.writeln(element.toString());
            }
        }
      }
    } catch (e) {
      stdout.writeln('Error: $e');
    }
  }

  late final String _prompt;
}

File: rust/src/bin/repl.rs
use anyhow::Result;
use ts_rust_zig_deez::lexer::lexer::{Lexer, Token};

fn main() -> Result<()> {

    std::io::stdin().lines().for_each(|line| {
        if let Ok(line) = line {
            let mut tokenizer = Lexer::new(line);

            while let Ok(token) = tokenizer.next_token() {
                println!("{} ", token);
                if let Token::Eof = token {
                    break;
                }
            }
        }
    });
    return Ok(());
}
